From b04b324c993d13bf5f8bf3bdfeb6be69569ccbe2 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:29:43 -0400
Subject: [net] openvswitch: avoid warnings in vport_from_priv

Message-id: <59c07d93ba98a536e3941b5a6901cf5b3e0ecf16.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94017
O-Subject: [RHEL7.1 net PATCH 10/50] openvswitch: avoid warnings in vport_from_priv
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit d0b4da137508db3d38998eae7f62c0f9699ee08c
Author: Daniele Di Proietto <daniele.di.proietto@gmail.com>
Date:   Mon Feb 3 14:07:43 2014 -0800

    openvswitch: avoid warnings in vport_from_priv

    This change, firstly, avoids declaring the formal parameter const,
    since it is treated as non const. (to avoid -Wcast-qual)
    Secondly, it cast the pointer from void* to u8*, since it is used
    in arithmetic (to avoid -Wpointer-arith)

    Signed-off-by: Daniele Di Proietto <daniele.di.proietto@gmail.com>
    Signed-off-by: Jesse Gross <jesse@nicira.com>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/vport.h b/net/openvswitch/vport.h
index 92137dd..47d252e 100644
--- a/net/openvswitch/vport.h
+++ b/net/openvswitch/vport.h
@@ -185,9 +185,9 @@ static inline void *vport_priv(const struct vport *vport)
  * the result of a hash table lookup.  @priv must point to the start of the
  * private data area.
  */
-static inline struct vport *vport_from_priv(const void *priv)
+static inline struct vport *vport_from_priv(void *priv)
 {
- return (struct vport *)(priv - ALIGN(sizeof(struct vport), VPORT_ALIGN));
+ return (struct vport *)((u8 *)priv - ALIGN(sizeof(struct vport), VPORT_ALIGN));
 }
 
 void ovs_vport_receive(struct vport *, struct sk_buff *,
-- 
1.7.1